### DISCUSSION TITLE
propSatisfies

### DISCUSSION THREAD

**Discussion by cjquines** [remeda/remeda#948] [Suggest a Function]
### Discussed in https://github.com/remeda/remeda/discussions/849

<div type='discussions-op-text'>

<sup>Originally posted by **viclafouch** September 11, 2024</sup>
I know it was discussed in https://github.com/remeda/remeda/issues/37 but I think `propEq` was one of the utility I used a lot when I used Ramda. 

Doing 

```
// Not this
const post = posts.find((item) => {
  return item.subject === 'toto'
})

// But this
const post = posts.find(R.propEq('subject', 'toto'))
```

Was very nice for Array.`find | filter | some | every..` etc...

Is there an equivalent in Remeda, or, if I want to build my own `propEq` in Typescript, do you have an idea ? 

Thanks a lot ! 

</div>

**Comment by eranhirsch** [comment]
`propSatisfies` with `isStrictEqual`/`isDeepEqual`/`isShallowEqual` would provide more control and better typing support and allow us to maintain fewer utilities.

```ts
const post = posts.find(propSatisfies("subject", isStrictEqual("toto")));
```
